src/stash/src/objects.rs

```diff
@@ -190,6 +190,8 @@ pub unsafe trait WireCompatible<T: prost::Message>: prost::Message + Default {
     }
 }
 
+unsafe impl WireCompatible<()> for () {}
```

Comment:
What's this needed for? If it's needed, the docs say the macro should be used.

Reply:
It is needed for `migrate_compat` to work on the `AUDIT_LOG_COLLECTION`:
```rust
AUDIT_LOG_COLLECTION.migrate_compat(tx, |_| updates).await?;
```

This collection has a value type of `()`. So without the `WireCompatible` impl, you get this error:
```
error[E0277]: the trait bound `(): WireCompatible<()>` is not satisfied
   --> src/stash/src/upgrade/v33_to_v34.rs:214:26
    |
214 |     AUDIT_LOG_COLLECTION.migrate_compat(tx, |_| updates).await?;
    |                          ^^^^^^^^^^^^^^ the trait `WireCompatible<()>` is not implemented for `()`
    |
    = help: the following other types implement trait `WireCompatible<T>`:
              <objects_v27::DatabaseKey as WireCompatible<v27_to_v28::objects_v28::DatabaseKey>>
              <objects_v27::DatabaseValue as WireCompatible<v27_to_v28::objects_v28::DatabaseValue>>
              <objects_v27::DefaultPrivilegesKey as WireCompatible<v27_to_v28::objects_v28::DefaultPrivilegesKey>>
              <objects_v27::DefaultPrivilegesValue as WireCompatible<v27_to_v28::objects_v28::DefaultPrivilegesValue>>
              <objects_v27::SchemaKey as WireCompatible<v27_to_v28::objects_v28::SchemaKey>>
              <objects_v27::SchemaValue as WireCompatible<v27_to_v28::objects_v28::SchemaValue>>
              <objects_v33::RoleKey as WireCompatible<v32_to_v33::objects_v32::RoleKey>>
              <objects_v33::RoleValue as WireCompatible<v32_to_v33::objects_v32::RoleValue>>
            and 16 others
note: required by a bound in `upgrade::<impl TypedCollection<K, V>>::migrate_compat`
   --> src/stash/src/upgrade.rs:148:20
    |
141 |     pub(crate) async fn migrate_compat<K2, V2>(
    |                         -------------- required by a bound in this associated function
...
148 |         V2: Data + WireCompatible<V>,
    |                    ^^^^^^^^^^^^^^^^^ required by this bound in `upgrade::<impl TypedCollection<K, V>>::migrate_compat`
```

Using the `wire_compatible!` macro doesn't work because it expects its arguments to be `ident`s, which `()` is not. I'm sure we could change the definition of the macro, but I would rather do that as a separate PR.

Reply:
https://github.com/MaterializeInc/materialize/pull/21323